convert2rhel/unit_tests/actions/report_test.py

```diff
@@ -1100,6 +1098,358 @@ def test_messages_summary_ordering(results, include_all_reports, expected_result
+                        "remediation": "move on",
+                        "variables": {},
+                    },
+                ),
+                "TestAction": dict(
+                    messages=[
+                        {
+                            "level": STATUS_CODE["WARNING"],
+                            "id": "WARNING_ID",
+                            "title": "Warning",
+                            "description": "Action warning",
+                            "diagnosis": "User warning",
+                            "remediation": "move on",
+                            "variables": {},
+                        }
+                    ],
+                    result={
+                        "level": STATUS_CODE["ERROR"],
+                        "id": "SECONDERROR",
+                        "title": "Error",
+                        "description": "Action error",
+                        "diagnosis": "User error",
+                        "remediation": "move on",
+                        "variables": {},
+                    },
+                ),
+            },
+            [
+                "{begin_fail}(ERROR) ErrorAction::ERROR - Error\n Description: Action error\n Diagnosis: User error\n Remediation: move on\n{end}",
+                "{begin_fail}(ERROR) TestAction::SECONDERROR - Error\n Description: Action error\n Diagnosis: User error\n Remediation: move on\n{end}",
+                "{begin_fail}(OVERRIDABLE) OverridableAction::OVERRIDABLE - Overridable\n Description: Action overridable\n Diagnosis: User overridable\n Remediation: move on\n{end}",
+                "{begin_fail}(SKIP) SkipAction::SKIP - Skip\n Description: Action skip\n Diagnosis: User skip\n Remediation: move on\n{end}",
+                "{begin_warning}(WARNING) SkipAction::WARNING_ID - Warning\n Description: Action warning\n Diagnosis: User warning\n Remediation: move on\n{end}",
+                "{begin_warning}(WARNING) OverridableAction::WARNING_ID - Warning\n Description: Action warning\n Diagnosis: User warning\n Remediation: move on\n{end}",
+                "{begin_warning}(WARNING) ErrorAction::WARNING_ID - Warning\n Description: Action warning\n Diagnosis: User warning\n Remediation: move on\n{end}",
+                "{begin_warning}(WARNING) TestAction::WARNING_ID - Warning\n Description: Action warning\n Diagnosis: User warning\n Remediation: move on\n{end}",
+            ],
+            True,
+        ),
+    ),
+)
+def test_report_to_txt(results, text_lines, file_exists, tmpdir, monkeypatch):
+    convert2rhel_txt_results = tmpdir.join("convert2rhel-pre-conversion.txt")
+    monkeypatch.setattr(report, "CONVERT2RHEL_TXT_RESULTS", str(convert2rhel_txt_results))
+
+    convert2rhel_txt_results = tmpdir.join("convert2rhel-pre-conversion.txt")
+
+    if file_exists:
+        convert2rhel_txt_results.write("test")
+
+    monkeypatch.setattr(report, "CONVERT2RHEL_TXT_RESULTS", str(convert2rhel_txt_results))
```

Comment:
I may be wrong here, but I think that you don't need to redo that part. Since your function will create the txt file in the end, it might as well be okay when you executed you don't have the file present. The idea, I think, is to test that your function will create the file and will put some contents in it. 

If I may suggest, I would remove this part here, as it is mostly the same thing that is being done above: 

```suggestion
```

And for your assertions, I would make that the file exists and has your content in it. Something like: 

```python
with open(convert2rhel_txt_results, "r") as file: 
    data = file.read()

# assert data == ...
```

Reply:
Or as I just saw below, you could just reuse `convert2rhel_txt_results.read()` instead of opening a new file handler for that.

Reply:
@r0x0d I see, it's probably copy&paste mistake (I used the first two lines for mocking the path in other tests. Now it's not used since I moved the call from the `summary`). I will rather remove the two lines above your mention:
```
convert2rhel_txt_results = tmpdir.join("convert2rhel-pre-conversion.txt")
monkeypatch.setattr(report, "CONVERT2RHEL_TXT_RESULTS", str(convert2rhel_txt_results))
```
Because there is missing the write to file part and we need that for check, if the file was overwritten. I will also add `assert` to check if the `test` is not present in the file.

Reply:
You could split the tests and make one that validates that use case... I think it you separate the cases for you testing it will make it more readable as well. 

For example: 
1. The good case where the file was written with the correct contents
2. One that there are no contents
3. File existed and it was overridden
4. etc...

Reply:
@r0x0d I'm not sure if it makes a big difference in this case (the 2. and 3. I would say are mostly the same, it doesn't matter (from my point of view) if the file exists and have content or just exists - I would say exists and has content is stronger).

So then the only difference is if the content is written or not.
- if not, the file doesn't exists. As far as I know `tmpdir.join("convert2rhel-pre-conversion.txt")` will just create the path, not actually create the file. Covers the 1.
- if I write the content, the file is created and has content. Covers 2. and 3.

Reply:
That's about right too. Maybe we should split the test in two then? one that covers 1 and the other that covers 2 and 3?

Reply:
If it makes the test more readable, I'll do that!

But still, the only difference will be in:
```
if file_exists:
        convert2rhel_txt_results.write("test")
```
So I'm still wondering if it isn't good moment for using `pytest.mark.parametrize` and turn off/on this part of code using the control variable.

Reply:
We could, that's definitely a path to follow, @hosekadam, but I think that putting conditionals inside tests is not as a good case as it is in the code.. 

Usually, the test case we want it to do only what is proposed for it to do, so in that case, parametrize this test case would be to add new variants to the same case and validate that it will work/pass for what the test is being proposed.

If you add the conditional to your test, then you are trying to assert two different behaviors instead of just one. My personal preference would be going to split the test and have them individually tested, even though, it becomes easier to debug and adapt/understand the test in the future if needed.

Reply:
Thanks for clarification! I don't know why I had in my mind that this is good idea :D It makes totally sense what you are saying.